src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,151 @@
+            String deadlineArguments = echo.substring(8).trim();
+            String[] splitArguments = deadlineArguments.split("/");
+            return new Deadline(isDone, splitArguments[0], splitArguments[1].substring(2).trim());
+        } else if (echo.startsWith("event")) {
+            String eventArguments = echo.substring(5).trim();
+            String[] splitArguments = eventArguments.split("/");
+            return new Event(isDone, splitArguments[0], splitArguments[1].substring(4).trim(),
+                    splitArguments[2].substring(2).trim());
+        } else {
+            throw(new InvalidInputException("      ☹ OOPS!!! File format seems weird"));
+        }
+    }
+
+    /**
+     * Returns boolean based on whether a corresponding input string has a non-task creation command.
+     * Attempts to carry out command if command is recognised, and returns true.
+     * Else returns false.
+     *
+     * @param echo Input String to be parsed for commands.
+     * @param tasks TaskList Object which encapsulates the current list of tasks.
+     * @param ui Ui Object to call Ui-related methods.
+     * @return Boolean if any tasks are recognised and carried out.
+     */
+    public static boolean parseCommands(String echo, TaskList tasks, Ui ui) {
+        if (echo.equals("list")) {
+            System.out.println("    OK, Here are the items in your list: ");
+            Ui.printArrayList(tasks.getList());
+            // put in loop to read the list
+            return true;
+        }
+
+        if (echo.startsWith("mark")) {
+            try {
+                int taskToModify = Integer.parseInt(echo.replaceAll("[^0-9]", ""));
+                tasks.markDone(taskToModify - 1);
+            } catch (Exception e) {
+                // TODO: handle exception
+                return false;
+            }
+            return true;
+        }
+
+        if (echo.startsWith("unmark")) {
+            try {
+                int taskToModify = Integer.parseInt(echo.replaceAll("[^0-9]", ""));
+                tasks.markUndone(taskToModify - 1);
+            } catch (Exception e) {
+                // TODO: handle exception
+                return false;
+            }
+            return true;
+        }
+
+        if (echo.startsWith("delete") || echo.startsWith("remove")) {
+            try {
+                int taskToModify = Integer.parseInt(echo.replaceAll("[^0-9]", ""));
+                tasks.removeTask(taskToModify - 1);
+                ui.printListNumber(tasks.getList());
+            } catch (Exception e) {
+                // TODO: handle exception
+                return false;
+            }
+            return true;
+        }
+
+        if (echo.startsWith("find")) {
+            String taskToFind = echo.substring(4).trim();
+            if (taskToFind.isEmpty()) {
+                System.out.println("    OPPS!! You can't find nothing");
+                return true;
+            }
```

Comment:
The string (has a minor typo, "OPPS" should be "OOPS") printed makes it look like an error, but `true` is returned which makes it seem like this is not an error. Perhaps it could be made clearer by either not printing "OOPS" in the message to show that this behaviour is intended.